add a formatting check for non-C++ scripts and code

phlptp opened the pull request:
<!--By submitting a pull request you are acknowledging that you have the right to license your code under the terms of this repositories license.  
Please review the [Contributing Guidelines](../CONTRIBUTING.md) for more details
Please make sure you fill the following sections. If this PR fixes an issue, please tag the issue number in the first section.
e.g. This fixes issue #123-->
### Summary
<!-- please finish the following statement -->
If merged this pull request will add a formatting check for some other formats besides code

### Proposed changes
<!-- Describe the highlights of the proposed changes here -->
- add format check and fix issues

phlptp commented:
@nightlark  how difficult would it be to make the formatting check the PR adds into something that would automatically generate a pull request like the cmake-format check?

nightlark commented:
If the checks update the files in-place it is easy, otherwise harder.

phlptp commented:
I think they do, but how would I tell for sure?

phlptp commented:
I thought about just fixing them manually but that would take a while and I guessing this will happen frequently so having it be a PR like the clang-format would be useful and save a lot of time.

nightlark commented:
Adding a step that runs `git status` after the pre-commit would probably show changed files.

phlptp commented:
```
Run git status
HEAD detached at pull/1265/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .ci/azure-pipelines.yml
	modified:   .ci/undef_blacklist.txt
	modified:   .circleci/asan_blacklist.txt
	modified:   .circleci/config.yml
	modified:   .circleci/tsan_suppression.txt
	modified:   .circleci/ubsan_suppression.txt
	modified:   .clang-format
	modified:   .clang-tidy
	modified:   .codacy.yml
	modified:   .githooks/pre-commit
	modified:   .github/PULL_REQUEST_TEMPLATE.md
	modified:   .github/SUPPORT.md
	modified:   .github/workflows/clang-format.yml
	modified:   .github/workflows/release-build.yml
	modified:   .github/workflows/run-clang-tidy-pr.sh
	modified:   .github/workflows/setup-macos-10.9-sdk.sh
	modified:   .github/workflows/swig-interface-gen.yml
	modified:   .travis.yml
	modified:   CHANGELOG.md
	modified:   CMakeLists.txt
	modified:   CONTRIBUTING.md
```

along with a bunch of other files.  So it looks like they are being modified

phlptp commented:
@nightlark  can you take a look at this, make sure I am not doing anything dumb here.  I want to make sure this is decent before changing out of draft since that will impact the PR's that get created.

nightlark commented on .github/workflows/formatting.yml:
```diff
@@ -0,0 +1,59 @@
+
+jobs:
+  pre-commit:
+    if: github.event.pull_request.draft != true && github.event.pull_request.head.repo.full_name == github.repository
```
This condition could get applied at the step level for staging changed files and creating the PR instead of the entire job.

phlptp replied:
Can you check the change, it doesn't seem to skip it in the right circumstances, not sure what I am doing wrong

nightlark replied:
Skipping the PR creation steps is the expected behavior right now?

phlptp replied:
right since it is a draft PR

nightlark commented on interfaces/CMakeLists.txt:
```diff
@@ -100,6 +100,7 @@ if(HELICS_ENABLE_SWIG OR swig_required)
             set(
                 CMAKE_PROGRAM_PATH
                 "C:/ProgramData/chocolatey/bin"
+                "C:/local/swigwin-4.0.2"
```
Is this just anticipating 4.0.2 coming out soon?

phlptp replied:
For some reason I thought it was out already,  I will revert this change.  I think we do need the develop branch of swig to support octave 5.2 though as an aside

phlptp commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5458829&bid=17892526)

phlptp commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5458829&bid=17892526)

phlptp commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5458829&bid=17892526)

phlptp commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5458829&bid=17892526)

phlptp commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5458829&bid=17892526)
